sql/service/src/main/scala/org/apache/spark/sql/service/internal/ServiceConf.scala

```diff
@@ -0,0 +1,489 @@
+      .createWithDefault(86400)
+
+  val THRIFTSERVER_THRIFT_HTTP_COOKIE_DOMAIN =
+    buildConf("spark.sql.thriftserver.thrift.http.cookie.domain")
+      .internal()
+      .doc("Domain for the HS2 generated cookies")
+      .stringConf
+      .createWithDefault(null)
+
+  val THRIFTSERVER_THRIFT_HTTP_COOKIE_PATH =
+    buildConf("spark.sql.thriftserver.thrift.http.cookie.path")
+      .internal()
+      .doc("Path for the SS2 generated cookies")
+      .stringConf
+      .createWithDefault("")
+
+  val THRIFTSERVER_THRIFT_HTTP_COOKIE_IS_SECURE =
+    buildConf("spark.sql.thriftserver.thrift.http.cookie.is.secure")
+      .internal()
+      .doc("Secure attribute of the SS2 generated cookie.")
+      .booleanConf
+      .createWithDefault(true)
+
+  val THRIFTSERVER_THRIFT_HTTP_COOKIE_IS_HTTPONLY =
+    buildConf("spark.sql.thriftserver.thrift.http.cookie.is.httponly")
+      .internal()
+      .doc("HttpOnly attribute of the SS2 generated cookie.")
+      .booleanConf
+      .createWithDefault(true)
+
+  val THRIFTSERVER_THRIFT_PORT = buildConf("spark.sql.thriftserver.thrift.port")
+    .internal()
+    .doc("Port number of SparkServer2 Thrift interface " +
+      "when spark.sql.thriftserver.transport.mode is 'binary'.")
+    .intConf
+    .createWithDefault(10000)
+
+  val THRIFTSERVER_THRIFT_SASL_QOP = buildConf("spark.sql.thriftserver.thrift.sasl.qop")
+    .internal()
+    .doc("Sasl QOP value; set it to one of following values to enable higher levels of" +
+      "protection for SparkServer2 communication with clients." +
+      "Setting hadoop.rpc.protection to a higher level than SparkServer2 does not" +
+      "make sense in most situations. HiveServer2 ignores hadoop.rpc.protection in favor" +
+      "of spark.sql.thriftserver.thrift.sasl.qop." +
+      "  \"auth\" - authentication only (default)" +
+      "  \"auth-int\" - authentication plus integrity protection" +
+      "  \"auth-conf\" - authentication plus integrity and confidentiality protection" +
+      "This is applicable only if SparkServer2 is configured to use Kerberos authentication.")
+    .stringConf
+    .createWithDefault("auth")
+
+  val THRIFTSERVER_THRIFT_MIN_WORKER_THREADS =
+    buildConf("spark.sql.thriftserver.thrift.min.worker.threads")
+      .internal()
+      .doc("Minimum number of Thrift worker threads")
+      .intConf
+      .createWithDefault(5)
+
+  val THRIFTSERVER_THRIFT_MAX_WORKER_THREADS =
+    buildConf("spark.sql.thriftserver.thrift.max.worker.threads")
+      .internal()
+      .doc("Maximum number of Thrift worker threads")
+      .intConf
+      .createWithDefault(500)
+
+
+  val THRIFTSERVER_THRIFT_LOGIN_BEBACKOFF_SLOT_LENGTH =
+    buildConf("spark.sql.thriftserver.thrift.exponential.backoff.slot.length")
+      .internal()
+      .doc("Binary exponential backoff slot time for Thrift clients during login to SparkServer2," +
+        "for retries until hitting Thrift client timeout")
+      .timeConf(TimeUnit.MILLISECONDS)
```

Comment:
`.timeConf(TimeUnit.MILLISECONDS)` -> `intConf `?

Reply:
> `.timeConf(TimeUnit.MILLISECONDS)` -> `intConf `?

In hive is Time Type